### DISCUSSION TITLE
Multi-function GPIO button.

### DISCUSSION THREAD

**Discussion by SubLim3JS** [MiczFlor/RPi-Jukebox-RFID#1999] [General]
I did some searches and didn't see an existing thread for this. I would like to use 2 buttons but with multiple functions. With a quick press I want volume control and with a 2 second hold I want previous/next track. Has this already been done and I'm just not able to find the thread? If anyone can school me on this I would much appreciate it!

**Comment by SubLim3JS** [comment]
Hold please. I'll still likely need some support LoL but I did find a thread to start with. Hopefully I can use 2 buttons, tap for volume and hold for track skip and both at the same time to shutdown. I'll do some homework and play around. Please keep the discussion open for now in case I'm unable to get this going. 😊

https://github.com/MiczFlor/RPi-Jukebox-RFID/tree/master/components/gpio_control

GPIO 5 on PIN 29 (30 for Ground) is Volume Down.
GPIO 13 on PIN 33 (34 for Ground) is Volume Up.

**Comment by pabera** [comment]
This is very likely possible with the future3 branch.

https://rpi-jukebox-rfid.readthedocs.io/en/latest/userguide/gpioz.html

**Comment by SubLim3JS** [comment]
Can I experiment with the TwinButton: Six function beast on version 2 or is there specific back end scripting in version 3 required?

**Reply by SubLim3JS** [reply]
Must be backend scripts. I've been playing around with TwoButtonControl and still struggling with short and long presses.

I can short press either of two buttons to Pause/Play, Long press of button 1 "Prev" and long press of button 2 "Next" still doesn't behave right. It will Prev/Next track but then Pause. I have to short press again to continue playback.

Should I start experimenting with V3? If it helps the community I'm happy to help where I can. I'm a novice at multiple Pi projects but known to discover things to help everyone. Here is the latest configuration I've used.

[PauseAndPrev]
enabled : True
type : Button
pin : 5
pull_up_down : pull_up
hold_time : 1
hold_mode : SecondFunc
functioncall : functionCallPlayerPause
functioncall2 : functionCallPlayerPrev

[PauseAndNext]
enabled : True
type : Button
pin : 13
pull_up_down : pull_up
hold_time : 1
hold_mode : SecondFunc
functioncall : functionCallPlayerPause
functioncall2 : functionCallPlayerNext

**Reply by pabera** [reply]
Have you tried version 3? It's a functional release, just with a few missing features here and there. The [TwinButton](https://rpi-jukebox-rfid.readthedocs.io/en/latest/userguide/gpioz.html#twinbutton-six-function-beast) should be working, as far as I know.

**Reply by AlvinSchiller** [reply]
This is now fixed with https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2367 for v2.8.0.
The behavior of `hold_mode` `SecondFunc` and `SecondFuncRepeat` has changed. The secondary function is no longer triggered additionally to the primary function.
Now its called exclusively if `hold_time` is reached. The primary function will only be triggered if the button is pressed shorter then `hold_time`! Existing configurations may need to adapt to this.

If you still encounter the execution of `functioncall ` after the hold, try setting the `antibouncehack` to `True`.
